src/main/java/com/aws/greengrass/clientdevices/auth/metrics/MetricsEmitter.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.clientdevices.auth.metrics;
+
+import com.aws.greengrass.telemetry.PeriodicMetricsEmitter;
+import com.aws.greengrass.telemetry.impl.Metric;
+import com.aws.greengrass.telemetry.impl.MetricFactory;
+
+import java.util.List;
+import javax.inject.Inject;
+
+public class MetricsEmitter extends PeriodicMetricsEmitter {
```

Comment:
I think that we'd like to modify the Nucleus in order to support this, however that will require bumping our minimum nucleus version, which we certainly want to avoid.

If we do not change the nucleus to expose a handle in TelemetryAgent to register external metric emitters, we will need similar logic in here in order to call `emitMetrics` on a schedule (1 hour default).

Reply:
I'm OK to commit this without a change given that this is the first step. Actual metric emission (call emitMetrics on a timer) can come in the next PR.

Reply:
Are we looking to add something along these lines: [https://github.com/aws-greengrass/aws-greengrass-nucleus/blob/221c28bde96ee7c9aec853ed81fffeee0376ae43/src/main/java/com/aws/greengrass/telemetry/TelemetryAgent.java#L164](url)

Is this something we want to update right now or later on?

Reply:
In a different PR.

Reply:
https://github.com/aws-greengrass/aws-greengrass-nucleus/blob/221c28bde96ee7c9aec853ed81fffeee0376ae43/src/main/java/com/aws/greengrass/telemetry/TelemetryAgent.java#L188-L192